### PR TITLE
[cu-38vhm86] Add Pablo to Picasso

### PR DESCRIPTION
**Pull request by vimukthi-git**
Add Pablo to Picasso runtime

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-38vhm86 Add Dex Pallet (refactor) to the Picasso runtime](https://app.clickup.com/t/38vhm86)

File: code/parachain/runtime/picasso/src/lib.rs [hunk 1]
```diff
@@ -253,6 +257,39 @@ impl assets_registry::Config for Runtime {
 	type WeightInfo = weights::assets_registry::WeightInfo<Runtime>;
 }
 
+parameter_types! {
+	pub PabloPalletId: PalletId = PalletId(*b"pal_pblo");
+	pub TWAPInterval: u64 = (MILLISECS_PER_BLOCK as u64) * 10;
+	pub MaxStakingRewardsPools: u32 = 10;
```
Comment (vimukthi-git) [inline code/parachain/runtime/picasso/src/lib.rs:263]: Remove unnecessary configurations related to already removed staking dependency

File: code/parachain/runtime/picasso/src/migrations.rs [hunk 2]
```diff
@@ -45,6 +48,110 @@ impl OnRuntimeUpgrade for TechCollectiveRenameMigration {
...
+		if !Pablo::pool_count().is_zero() {
+			return 0
+		}
+
+		pools.iter().for_each(|pool_creation_input| {
+			Pablo::do_create_pool(
+				pool_creation_input.init_config.to_owned(),
+				Some(pool_creation_input.lp_token),
+			)
+			.expect("Pool config is valid; QED");
```
Comment (andyjsbell) [inline code/parachain/runtime/picasso/src/migrations.rs:101]: What is the expected behaviour for a migration if this assert is met?

File: code/parachain/runtime/picasso/src/migrations.rs [hunk 3]
```diff
@@ -45,6 +48,110 @@ impl OnRuntimeUpgrade for TechCollectiveRenameMigration {
...
+			}
+		}
+	}
+
+	/// Adds pools to Pablo Storage
+	///
+	/// Expects a vec of (pool_init_config, pool_lp_token_id)
+	pub fn add_initial_pools_to_storage(pools: Vec<PoolCreationInput>) -> Weight {
+		if !Pablo::pool_count().is_zero() {
+			return 0
```
Comment (andyjsbell) [inline code/parachain/runtime/picasso/src/migrations.rs:93]: Aren't we reading from storage here?

File: code/parachain/runtime/picasso/src/weights/pablo.rs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+	// Storage: Pablo PriceCumulativeState (r:1 w:1)
+	fn swap() -> Weight {
+		(285_540_000 as Weight)
+			.saturating_add(T::DbWeight::get().reads(12 as Weight))
+			.saturating_add(T::DbWeight::get().writes(9 as Weight))
+	}
+
+	fn do_create_pool() -> Weight {
+		10_000
+    }
```
Comment (PoisonPhang) [inline code/parachain/runtime/picasso/src/weights/pablo.rs:90]: We'll want to make sure we generate weights for this PR

**Review by PoisonPhang — APPROVED** [review]
LGTM :rocket: 

Do make sure weights are generated before release

File: code/parachain/runtime/picasso/src/weights/pablo.rs (hunk 4 above)
Comment (vimukthi-git) [reply code/parachain/runtime/picasso/src/weights/pablo.rs:90]: We will do this for the release as per the [RFC](https://github.com/ComposableFi/composable/blob/main/rfcs/0011-releases.md#43-act) sir

**Review by andyjsbell — APPROVED** [review]
Thanks @vimukthi-git

**Review by KaiserKarel — APPROVED** [review]

**Pull request merged**